Zend/zend_language_parser.y

```diff
@@ -231,6 +231,8 @@ static YYSIZE_T zend_yytnamerr(char*, const char*);
 %token T_COALESCE        "'??'"
 %token T_POW             "'**'"
 %token T_POW_EQUAL       "'**='"
+%token T_AMPERSAND_NOT_FOLLOWED_BY_VAR_OR_VARARG "'&'"
+%token T_AMPERSAND_FOLLOWED_BY_VAR_OR_VARARG     "'&''"
```

Comment:
IIRC @iluuu1994 said this was needed for the tokens to be different

Reply:
Bison emits a warning if they're the same but it would be better for sure to avoid confusing error messages. Not sure if the warning can be suppressed.

Reply:
It does show up as `'&''` in the output see an extract:
```
    "'&'" (287) 80 272 273 283 284 390
    "'&''" (288) 79 184 391
```

Reply:
I mean in parse errors. We can't have it appear that way in parse errors, otherwise it doesn't matter much.

Reply:
Yes it does, I don't know how exactly the parse errors are generated but I'll look into that.

Reply:
You should be able to special case the token in zend_yytnamerr to fix parse errors: https://heap.space/xref/php-src/Zend/zend_language_parser.y?r=724230d9#1501

Something like this I think:

```
if (strcmp(toktype, "'&''") == 0) {
	if (yyres) {
		yystpcpy(yyres, "'&'");
	}
	return sizeof("'&'")-1;
}
```